pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -1111,10 +1112,25 @@ func (dm *DockerManager) ExecInContainer(containerID kubecontainer.ContainerID,
 		return fmt.Errorf("container not running (%s)", container.ID)
 	}
 
-	return dm.execHandler.ExecInContainer(dm.client, container, cmd, stdin, stdout, stderr, tty)
+	return dm.execHandler.ExecInContainer(dm.client, container, cmd, stdin, stdout, stderr, tty, resize)
 }
 
-func (dm *DockerManager) AttachContainer(containerID kubecontainer.ContainerID, stdin io.Reader, stdout, stderr io.WriteCloser, tty bool) error {
+func (dm *DockerManager) AttachContainer(containerID kubecontainer.ContainerID, stdin io.Reader, stdout, stderr io.WriteCloser, tty bool, resize <-chan term.Size) error {
+	// Have to start this before the call to client.AttachToContainer because client.AttachToContainer is a blocking
+	// call :-( Otherwise, resize events don't get processed and the terminal never resizes.
+	go func() {
+		defer utilruntime.HandleCrash()
+		for {
+			size, ok := <-resize
+			if !ok {
+				return
+			}
+			if err := dm.client.ResizeContainerTTY(containerID.ID, int(size.Height), int(size.Width)); err != nil {
```

Comment:
Maybe? Someone could theoretically write some go code that sets up an attach session and then repeatedly sends resize dimensions that always cause the `ResizeContainerTTY` to fail, I suppose. I can make sure there's a `>= 1` check in the code somewhere, at the very least. Beyond that, I guess we have to decide if we log this or not.